Data/Functor/Foldable.hs

```diff
@@ -175,18 +334,74 @@ distPara = distZygo embed
 distParaT :: (Corecursive t, Comonad w) => (forall b. Base t (w b) -> w (Base t b)) -> Base t (EnvT t w a) -> EnvT t w (Base t a)
 distParaT t = distZygoT embed t
 
+-- | A recursive datatype which can be rolled up one recursion layer at a time.
+--
+-- For example, a value of type @ListF a [a]@ can be rolled up into a @[a]@.
+-- This @[a]@ can then be used in a 'Cons' to construct another @List F a [a]@,
+-- which can be rolled up as well, and so on.
+--
+-- Typically, 'Corecursive' types also have a 'Recursive' instance, in which
+-- case 'embed' and 'project' are inverses.
 class Functor (Base t) => Corecursive t where
+  -- | Roll up a single recursion layer.
+  --
+  -- >>> embed (Cons 1 [2,3])
+  -- [1,2,3]
   embed :: Base t t -> t
+
+  -- | A generalization of 'unfoldr'. The starting seed is expanded into a base
+  -- functor whose recursive positions contain more seeds, which are themselves
+  -- expanded, and so on.
+  --
+  -- > -- |
+  -- > -- >>> enumFromTo 1 4
+  -- > -- [1,2,3,4]
+  -- > enumFromTo :: Int -> Int -> [Int]
+  -- > enumFromTo lo hi = ana go lo where
+  -- >   go :: Int -> ListF Int Int
+  -- >   go i = if i > hi then Nil else Cons i (i+1)
   ana
     :: (a -> Base t a) -- ^ a (Base t)-coalgebra
     -> a               -- ^ seed
     -> t               -- ^ resulting fixed point
   ana g = a where a = embed . fmap a . g
 
+  -- | A variant of 'ana' in which recursive positions may contain a sub-tree
+  -- instead of a seed.
+  --
+  -- Useful for short-circuiting the remainder of the unfolding:
+  --
+  -- > -- |
+  -- > -- >>> mergeSortedLists [1,4,6,9] [2,4,6,7,10]
+  -- > -- [1,2,4,4,6,6,7,9,10]
+  -- > mergeSortedLists :: [Int] -> [Int] -> [Int]
+  -- > mergeSortedLists xs1 xs2 = apo mergeSortedListsF (xs1,xs2)
+  -- >
+  -- > mergeSortedListsF :: ([Int],[Int]) -> ListF Int (Either [Int] ([Int],[Int]))
+  -- > mergeSortedListsF ([], []) = Nil
+  -- > mergeSortedListsF ([], x:xs2) = Cons x $ Left xs2
+  -- > mergeSortedListsF (x:xs1, []) = Cons x $ Left xs1
+  -- > mergeSortedListsF (x1:xs1, x2:xs2)
+  -- >   | x1 <= x2  = Cons x1 $ Right (xs1, x2:xs2)
```

Comment:
this is where `recursion-schemes` fall short, we are folding over lists, but there's a "tree" of choices (go left or right), and we make it here.

Reply:
something between 1 and 3.